collated/main/A0138411N.md

```diff
@@ -0,0 +1,877 @@
+        lastListing = listing;
+    }
+```
+###### /main/java/seedu/ggist/model/ModelManager.java
+``` java
+    /**
+     * Updates filtered list to show based on last shown listing choice
+     */
+    public void updateListing() {
+        if (lastListing == null) {
+            updateFilteredListToShowDate(today);
+        } else if (lastListing.equals("")) {
+            updateFilteredListToShowAllUndone();
+        } else if (lastListing.equals("done")) {
+            updateFilteredListToShowAllDone();
+        } else if (lastListing.equals("all")){
+            updateFilteredListToShowAll();
+        } else if (TaskDate.isValidDateFormat(lastListing)) {
+            updateFilteredListToShowDate(lastListing);
+        }
+    }
+     
+    @Override
+    public UnmodifiableObservableList<ReadOnlyTask> getFilteredTaskList() {
+        return getSortedTaskList();
+    }
+    
+    @Override
+    public UnmodifiableObservableList<ReadOnlyTask> getSortedTaskList() {
+        Comparator<Task> compareDateTime = new Comparator<Task>(){
+            public int compare (Task t1, Task t2){
+                    
+                if (t1.getStartDateTime().equals(t2.getStartDateTime())
+                    && (t1.getEndDateTime().equals(t2.getEndDateTime()))) {
+                    return t1.getTaskName().taskName.compareTo(t2.getTaskName().taskName);
+                } 
+                
+                if (t1.getStartDateTime().before(t2.getStartDateTime())) {
+                    return -1;
+                } else if (t1.getStartDateTime().after(t2.getStartDateTime())) {
+                    return 1;
+                } 
+                
+                if (t1.getEndDateTime().before(t2.getEndDateTime())) {
+                    return -1;
+                } else if (t1.getEndDateTime().after(t2.getEndDateTime())) {
+                    return 1;
+                }
+                
+                return 0;
+            }
+        };
+        return new UnmodifiableObservableList<>(new SortedList(filteredTasks, compareDateTime));
+    }
+```
+###### /main/java/seedu/ggist/model/ModelManager.java
+``` java
+    @Override
+    public void updateFilteredListToShowDate(String keywords){
+        updateFilteredTaskList(new PredicateExpression(new DateQualifier(keywords)));
+    }
+
+    private void updateFilteredListToShowDate(Expression expression) {
+        filteredTasks.setPredicate(expression::satisfies);
+    }
+```
+###### /main/java/seedu/ggist/model/ModelManager.java
+``` java
+    private class DateQualifier implements Qualifier {
+        private String taskDateKeyWords;
+
+        DateQualifier(String taskDateKeyWords) {
+            this.taskDateKeyWords = taskDateKeyWords;
+        }
+
+        @Override
+        public boolean run(ReadOnlyTask task) {
+            if (taskDateKeyWords == null) {
+                return true;
+            }
+            return ((taskDateKeyWords.equalsIgnoreCase(task.getStartDate().toString()) || 
```

Comment:
is there a better way to manage this return statement? it seems to be doing a LOT of work :-P